packages/connectors/src/DataConnector.ts

```diff
@@ -0,0 +1,54 @@
+import { ConnectorConfigValue, ConnectorRuntimeContext, Dictionary } from './Connector.Shared';
+
+export interface DataConnector {
+    getPage(config: PageConfig, context: Dictionary): Promise<DataPage>;
+    getModel(context: Dictionary): Promise<DataModel>;
+
+    getConfigurationOptions(): ConnectorConfigValue[] | null;
+    getCapabilities(): DataConnectorCapabilities;
+}
+
+// TODO: add methods once defined
+export type DataConnectorRuntimeContext = ConnectorRuntimeContext;
+
+export type DataConnectorCapabilities = {
+    filtering: boolean;
+    sorting: boolean;
+    model: boolean;
+};
+
+export class DataModel {
+    properties: DataModelProperty[];
+}
+
+export class DataModelProperty {
+    name: string;
+    type: 'number' | 'boolean' | 'singleLine' | 'multiLine' | 'date';
```

Comment:
'singleLineText' and `multiLineText' ?

Reply:
we have decided to leave as it is for now